python/gstaichi/lang/kernel_impl.py

```diff
@@ -336,6 +336,12 @@ def _process_args(self: "Func | Kernel", is_func: bool, args: tuple[Any, ...], k
         arg_str = ", ".join(map(str, args))
         expected_str = ", ".join(f"{arg.name} : {arg.annotation}" for arg in self.arg_metas)
         msg = f"Too many arguments. Expected ({expected_str}), got ({arg_str})."
+        for i in range(len_args):
+            if i < len(self.arg_metas):
+                print(i, self.arg_metas[i].name, type(args[i]))
+            else:
+                print(i, "<out of arg metas>", type(args[i]))
+        print("in function", self.func)
         raise GsTaichiSyntaxError(msg)
```

Comment:
You are mixing prints are exception. I don't think it is right. I would suggest formatting a single clean error message containing all the information, possibly with multiple lines and bullet points.